ts/src/header-validator/source.test.ts

```diff
@@ -2544,6 +2559,223 @@ const testCases: TestCase[] = [
+        msg: 'must be in the range [1, 4294967295]',
+      },
+    ],
+  },
+  {
+    name: 'attribution-scope-limit-not-integer',
+    json: `{
+      "destination": "https://a.test",
+      "attribution_scope_limit": 1.5
+    }`,
+    expectedErrors: [
+      {
+        path: ['attribution_scope_limit'],
+        msg: 'must be an integer',
+      },
+    ],
+  },
+  {
+    name: 'attribution-scope-limit-exceeds-max',
+    json: `{
+      "destination": "https://a.test",
+      "attribution_scope_limit": 4294967296
+    }`,
+    expectedErrors: [
+      {
+        path: ['attribution_scope_limit'],
+        msg: 'must be in the range [1, 4294967295]',
+      },
+    ],
+  },
+  {
+    name: 'empty-attribution-scopes-with-limit',
+    json: `{
+      "destination": "https://a.test",
+      "attribution_scope_limit": 3,
+      "attribution_scopes": []
+    }`,
+    expectedErrors: [
+      {
+        path: ['attribution_scopes'],
+        msg: 'length must be in the range [1, 3]',
+      },
+    ],
+  },
+  {
+    name: 'missing-attribution-scope-limit-attribution-scopes',
+    json: `{
+      "destination": "https://a.test",
+      "attribution_scopes": ["1", "2"]
+    }`,
+    expectedErrors: [
+      {
+        path: ['attribution_scopes'],
+        msg: 'length must be in the range [0, 0]',
+      },
+    ],
+  },
+  {
+    name: 'missing-attribution-scope-limit-max-event-states',
+    json: `{
+      "destination": "https://a.test",
+      "max_event_states": 5
+    }`,
+    expectedErrors: [
+      {
+        path: ['max_event_states'],
+        msg: 'non-default max_event_states when attribution_scope_limit is not set',
+      },
+    ],
+  },
+  {
+    name: 'max-event-states-negative',
+    json: `{
+      "destination": "https://a.test",
+      "attribution_scope_limit": 1,
+      "max_event_states": -1
+    }`,
+    expectedErrors: [
+      {
+        path: ['max_event_states'],
+        msg: 'must be in the range [1, Infinity]',
+      },
+    ],
+  },
+  {
+    name: 'max-event-states-zero',
+    json: `{
+      "destination": "https://a.test",
+      "attribution_scope_limit": 1,
+      "max_event_states": 0
+    }`,
+    expectedErrors: [
+      {
+        path: ['max_event_states'],
+        msg: 'must be in the range [1, Infinity]',
+      },
+    ],
+  },
+  {
+    name: 'max-event-states-not-integer',
+    json: `{
+      "destination": "https://a.test",
+      "attribution_scope_limit": 1,
+      "max_event_states": 1.5
+    }`,
+    expectedErrors: [
+      {
+        path: ['max_event_states'],
+        msg: 'must be an integer',
+      },
+    ],
+  },
+  {
+    name: 'attribution-scopes-size-exceeds-attribution-scope-limit',
+    json: `{
+      "destination": "https://a.test",
+      "attribution_scope_limit": 1,
+      "attribution_scopes": ["1", "2"]
+    }`,
+    expectedErrors: [
+      {
+        path: ['attribution_scopes'],
+        msg: 'length must be in the range [1, 1]',
```

Comment:
We may want to indicate that the second `1` in this message is derived from the scope limit field.